### DISCUSSION TITLE
transposeObject

### DISCUSSION THREAD

**Discussion by silveltman** [remeda/remeda#952] [Suggest a Function]
similar to https://radash-docs.vercel.app/docs/object/map-entries

Seems to make sense because just like Object.values and Object.keys, Object.entries does also exist in regular javascript.

**Comment by eranhirsch** [comment]
We have `entries()` and `map()`. We tend not to add very simple compositions of other utils if they don't benefit at runtime or can provide better typing than using the composition itself.

**Comment by cjquines** [comment]
i think we could add better typing for `mapEntries` than the composition `entries`, `map`, `fromEntries`

in particular, the same arguments can be made for `mapEntries`, `mapKeys`, and `mapValues`; if we have the latter two, i don't see why not to have `mapEntries`

**Comment by eranhirsch** [comment]
If the request is to also build the object after the iteration i agree and think it is a good addition to the library, i think the best api for this would be similar to pullObject, and a good name would be transformObject.
It exists in hack as Dict\pull_with_keys

**Comment by cjquines** [comment]
i believe that is the request! i'll move this to good first issue

**Comment by cjquines** [comment]
thinking about this more... not sure if something like `mapEntries(data, key => key, value => value)` is the desired API, since your function can't refer to the `key` and `value` at the same time?

**Comment by eranhirsch** [comment]
Both callbacks should take the key and value (and the object itself). The order could be flipped so that you can write a simpler 1-param function in most cases:

```ts
transformObject<T>(
  data: T, 
  keyMapper: (key: ..., value: ..., data: T) => K
  valueMapper: (value: ..., key: ..., data: T) => V
): ...
```

**Comment by eranhirsch** [comment]
@cjquines, what do you think about `transposeObject`? ~~Transform is already taken by `transform,` which takes an object and maps its properties~~ (EDIT: Actually I was wrong; it's called `evolve`).

**Comment by cjquines** [comment]
i think the name that people will look for is `mapObject` or `mapEntries`
